pkg/tsdb/influxdb/influxql/response_parser.go

```diff
@@ -60,35 +69,34 @@ func parseJSON(buf io.Reader) (models.Response, error) {
 }
 
 func transformRows(rows []models.Row, query models.Query) data.Frames {
-	// pre-allocate frames - this can save many allocations
-	cols := 0
+	// Create a map for faster column name lookups
+	columnToLowerCase := make(map[string]string)
 	for _, row := range rows {
-		cols += len(row.Columns)
+		for _, column := range row.Columns {
+			columnToLowerCase[column] = strings.ToLower(column)
+		}
 	}
-	frames := make([]*data.Frame, 0, len(rows)+cols)
+
+	// Preallocate for the worst-case scenario
+	frames := make([]*data.Frame, 0, len(rows)*len(rows[0].Columns))
```

Comment:
I am not sure if this change is wise. I need feedback on this one.

Reply:
hmm... what's the impact from your benchmark runs between the old version and this?

Reply:
```
pkg: github.com/grafana/grafana/pkg/tsdb/influxdb/influxql
                │    1.txt    │               15.txt               │
                │   sec/op    │   sec/op     vs base               │
ParseBigJson-10   528.0m ± 5%   505.0m ± 2%  -4.37% (p=0.003 n=10)

                │    1.txt     │               15.txt                │
                │     B/op     │     B/op      vs base               │
ParseBigJson-10   580.6Mi ± 0%   580.4Mi ± 0%  -0.04% (p=0.000 n=10)

                │    1.txt    │            15.txt             │
                │  allocs/op  │  allocs/op   vs base          │
ParseBigJson-10   10.12M ± 0%   10.12M ± 0%  ~ (p=0.105 n=10)
```